### ISSUE TITLE
Execution failed for task ':app:minifyReleaseWithR8 post 'Flutter 3.24 + Packages Upgrade' update

### ISSUE THREAD

**Issue by Yash-bibli** [FlutterFlow/flutterflow-issues#4602]
### Can we access your project?

- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

Post yesterdays version update, the app has stopped building while Flutterflow doesn't show any error. Have been seeing the error related 'Build failed' related to minifyReleaseWithR8.

### Expected Behavior

Since there is no error, we have also managed the dependencies of external code properly. So not sure why build is failing. This has stared post the update 'Flutter 3.24 + Packages Upgrade' as communicated by Flutterflow

### Steps to Reproduce

Push the build to Google Play. It shows failed logs. Here is the screenshot of the logs.
<img width="829" alt="Screenshot 2024-10-24 at 6 28 45 PM" src="https://github.com/user-attachments/assets/a91f00b0-2499-48cd-8f8a-5ff210b9b405">

### Reproducible from Blank

- [ ] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

IT4slsrlw4xNpbwF1s7uKcJB/SEWH2IgR+dJsctFb0ggGJTsBu0PPMmkckpDQu2gaVNcEGKKglsyw+/FjvLhKMY7NQ6cGohC+s8Jew3Oe2i6br7RDbuWXEF7M51MIGqXyruRhgpABPBvS1IZ3V2MIuvKdArDFOu/Zwh9f6PHaOI=

### Visual documentation

https://github.com/user-attachments/assets/5e7671c5-7960-4948-98f0-22ad3998307c

### Environment

```markdown
- FlutterFlow version: 5.08+
- Platform: Web
- Browser name and version: Chrome 129.0.6668.91
- Operating system and version affected: mac OS 14.2.1 (23C71)
```

### Additional Information

We are not able to deploy the project as the build is failing. Have to release some features asap.

**Comment by Alezanello** [comment]
Hello,

I tested your project by downloading the code and compiling it locally on an Android physical device. The compilation was successful, and the app launched without issues. However, I encountered several exceptions when attempting to build the release version.

It appears that some of your dependencies require a lower version of Kotlin than the one currently being used. Specifically, many modules were compiled with Kotlin 1.8.0, but the expected version is 1.6.0. I recommend reviewing your project's dependencies and ensuring that all packages are compatible with the correct Kotlin version.

Below are the specific exceptions that were thrown during the build process:

```
e: C:/<User>/.gradle/caches/transforms-3/b12191bbe4b20fbcbb94aadaa1eed12f/transformed/fragment-1.7.1-api.jar!/META-INF/fragment_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

e: C:/<User>/.gradle/caches/transforms-3/88153fa48a3aa49260e5f2b259cdb002/transformed/jetified-activity-1.8.1-api.jar!/META-INF/activity_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

e: C:/<User>/.gradle/caches/transforms-3/39bb1d2d77356f50ebb6d6c3469591ea/transformed/jetified-lifecycle-livedata-core-ktx-2.7.0-api.jar!/META-INF/lifecycle-livedata-core-ktx_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

e: C:/<User>/.gradle/caches/transforms-3/64a01fdf6c5c1f7d28b962d276b65621/transformed/lifecycle-livedata-2.7.0-api.jar!/META-INF/lifecycle-livedata_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

e: C:/<User>/.gradle/caches/transforms-3/a9074af2d5fe833299de6ed0e0c3d1f5/transformed/lifecycle-viewmodel-2.7.0-api.jar!/META-INF/lifecycle-viewmodel_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

e: C:/<User>/.gradle/caches/transforms-3/cb00672285607ee499536fc80d510916/transformed/lifecycle-livedata-core-2.7.0-api.jar!/META-INF/lifecycle-livedata-core_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

e: C:/<User>/.gradle/caches/transforms-3/9cfdbef56e2b6f3a5f7d10c4ecac4604/transformed/jetified-lifecycle-viewmodel-savedstate-2.7.0-api.jar!/META-INF/lifecycle-viewmodel-savedstate_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

e: C:/<User>/.gradle/caches/transforms-3/b608ec0ce7126ec049b4f69c38fede2e/transformed/jetified-core-ktx-1.13.1-api.jar!/META-INF/core-ktx_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

e: C:/<User>/.gradle/caches/transforms-3/64dd606935fc6c97ab951bea72ee7d59/transformed/core-1.13.1-api.jar!/META-INF/core_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

e: C:/<User>/.gradle/caches/transforms-3/f24ade03bba0df4130d94361f75995aa/transformed/lifecycle-runtime-2.7.0-api.jar!/META-INF/lifecycle-runtime_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

e: C:/<User>/.gradle/caches/transforms-3/be7a46c16735152e0fe9a8e9c22df56c/transformed/jetified-lifecycle-process-2.7.0-api.jar!/META-INF/lifecycle-process_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

e: C:/<User>/.gradle/caches/modules-2/files-2.1/androidx.lifecycle/lifecycle-common/2.7.0/85334205d65cca70ed0109c3acbd29e22a2d9cb1/lifecycle-common-2.7.0.jar!/META-INF/lifecycle-common.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

e: C:/<User>/.gradle/caches/transforms-3/c10dc1f340a2476184aa723321f4613f/transformed/jetified-savedstate-1.2.1-api.jar!/META-INF/savedstate_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

e: C:/<User>/.gradle/caches/transforms-3/906794553d473cd91abd947374b81672/transformed/jetified-window-1.2.0-api.jar!/META-INF/window_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

e: C:/<User>/.gradle/caches/transforms-3/a0731ee104c61e5fa35da4995d2ce940/transformed/jetified-window-java-1.2.0-api.jar!/META-INF/window-java_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

e: C:/<User>/.gradle/caches/transforms-3/bca858adc701db7e291351fa8710682b/transformed/jetified-kotlinx-coroutines-core-jvm-1.7.1.jar!/META-INF/kotlinx-coroutines-core.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

e: C:/<User>/.gradle/caches/transforms-3/65858d769a1ece42bb957057e0ee7e23/transformed/jetified-kotlinx-coroutines-android-1.7.1.jar!/META-INF/kotlinx-coroutines-android.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

e: C:/<User>/.gradle/caches/transforms-3/740185fb53c8e058a427227460e5de22/transformed/jetified-kotlin-stdlib-1.8.22.jar!/META-INF/kotlin-stdlib-jdk7.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

e: C:/<User>/.gradle/caches/transforms-3/740185fb53c8e058a427227460e5de22/transformed/jetified-kotlin-stdlib-1.8.22.jar!/META-INF/kotlin-stdlib.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

e: C:/<User>/.gradle/caches/transforms-3/740185fb53c8e058a427227460e5de22/transformed/jetified-kotlin-stdlib-1.8.22.jar!/META-INF/kotlin-stdlib-jdk8.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.

e: C:/<User>/.gradle/caches/transforms-3/bad04b841d0cf81f05572f060ce9ac63/transformed/jetified-kotlin-stdlib-common-1.8.22.jar!/META-INF/kotlin-stdlib-common.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.
```

<img src="https://uploads.linear.app/3241089d-736a-4422-900b-051e72accdbe/b0690e89-ae03-440d-9949-5d2145edae5f/f5ff25b3-5a3d-42f6-be13-42142dd983a3?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzMyNDEwODlkLTczNmEtNDQyMi05MDBiLTA1MWU3MmFjY2RiZS9iMDY5MGU4OS1hZTAzLTQ0MGQtOTk0OS01ZDIxNDVlZGFlNWYvZjVmZjI1YjMtNWEzZC00MmY2LWJlMTMtNDIxNDJkZDk4M2EzIiwiaWF0IjoxNzI5NzgxMTE3LCJleHAiOjMzMzAwMzQxMTE3fQ.0fQfh2Ffly6ChKts0iIh1bhLjnrw5FIkSjfAOtfV_bY " alt="image.png" width="926" height="972" />

**Comment by rapiddev-ff** [comment]
Hi, I write here because my report was closed because of duplicate. We check dependencies in our app and added latest versions. But the bug still exists

**Comment by Yash-bibli** [comment]
Just updating that we were able to push the app after changing the Kotlin version to 1.7.0 (Somehow 1.6.0 didn't work as well) in Flutterflow Interface.

**Comment by Yash-bibli** [comment]
We faced another issue after downgrading the Kotlin version - The app was crashing again and again, not sure if Kotlin was the issue or overall Flutterflow update. We checked it being related to Firestore and the recommendation related to 'Check Proguard rules' helped in resolving the crash issue. But will appreciate if these fixes can be done in the Flutterflow platform so that we don't have to find the workarounds.

![image](https://github.com/user-attachments/assets/7fdd3274-7405-49e8-b4e6-329713ce07bb)
